### ISSUE TITLE
All my DMs say the authenticity can't be verified

### ISSUE THREAD

**Issue by turt2live** [vector-im/element-web#14323]
I don't know why, and as a user I want to click a button to make the warning go away/be fixed

**Comment by turt2live** [comment]
this was on a new signin, so probably key backup?

**Comment by aljoscha** [comment]
<img width="421" alt="Screenshot 2020-07-04 at 08 46 46" src="https://user-images.githubusercontent.com/68551/86506769-e8847c80-bdd2-11ea-89cb-34cc9957c67c.png">

This happens to me too, along with a black shield. Hovering over the shield reveals the "authenticity can't be verified" message.

**Comment by 98WuG** [comment]
Can confirm, this happened to me this morning as well. Opened an existing session to see the "authenticity can't be guaranteed" message.

**Comment by 532910** [comment]
Both contacts have Cross-Signing and Key Backup configured on all devices.
All sessions are shown as verified with a green shield, that is shown on the top too: `Everyone in this room is verified`.
Both contacts are on the same HS.

Despite this, I see `The authenticity of this encrypted message can't be guaranteed on this device` for some of my and his new messages.

**Comment by jryans** [comment]
@uhoreg Hmm, is it expected to see this message for _new_ messages as well, or just those restored from key backup?

**Comment by 532910** [comment]
And I see this with different contacts for all criteria above.

**Comment by uhoreg** [comment]
> @uhoreg Hmm, is it expected to see this message for _new_ messages as well, or just those restored from key backup?

It could happen for new messages that use the same megolm session as old messages.  New megolm sessions should not show the message.

**Comment by dylangerdaly** [comment]
I can confirm I'm seeing this as well.

**Comment by uhoreg** [comment]
I thought that this issue contained an explanation for why this happens, but it doesn't :frowning_face: .

So, for everyone confused by why it's saying this: Normally, in encrypted rooms, messages are authenticated using the decryption key, based on the fact that the decryption key was received directly from the sender.  When you sign into a new device (or under some other situations), you may get the decryption key from key backup (keys stored encrypted on the server), or from key sharing (keys sent from your other devices).  When that happens, we lose the link to the original sender, and we don't currently store the fact that it was trusted in the first place, which is why it now says that the authenticity can't be verified.

**Comment by Avamander** [comment]
> When that happens, we lose the link to the original sender, and we don't currently store the fact that it was trusted in the first place, which is why it now says that the authenticity can't be verified.

How do I re-establish that link using Element?

> So the message is completely expected in certain situations (particularly for old messages when you log in a new device)

I get it for each message I send from my second device. They both have key backup set up and restored and are cross-signed.

**Comment by Avamander** [comment]
In general this message seems useless - it's undocumented and unactionable.

**Comment by jryans** [comment]
> How do I re-establish that link using Element?

There is no way to do so currently.

**Comment by jryans** [comment]
> In general this message seems useless - it's undocumented and unactionable.

I understand this frustration. I'll try raising this issue with our Product team internally.

**Comment by ara4n** [comment]
So the fact is that you only know for sure that your megolm keys are actually sent by the person who claimed to send them at the point you receive them directly for them.  If you receive them subsequently from your online backup, or re-shared by another party, then it's possible that they might be faked by an attacker who is colluding with your server to feed you fake messages.  There is work in progress to improve the trustworthiness of data of online backups, but the risk still applies to keyshared keys.

To improve the UX:

1. We could display a dialog when you restore keys from backup saying "encrypted history restored! (warning: we cannot prove your history has not been tampered with)" or words to that effect.

2. We could display an inline cell in the timeline saying "we cannot prove that history before this point has not been tampered with"

3. We could ignore the warning entirely, but this feels a bit irresponsible even though it's an edge case that you have an attacker trying to fake your past history to you.  For instance, an attacker on the server might deliberately inject compromising material into your history in order to frame you, and this warning confirms that currently you can't trust your history 100% to be free from that, when keys are restored from backup.

Whatever, keyshared msgs would still have the individual shield warning.

**Comment by Avamander** [comment]
Okay, but in the case of new messages, not backups. What's stopping from trusting those? Did I not cross-sign the devices to verify exactly this?

**Comment by uhoreg** [comment]
If it's on new messages, then it's a different issue (https://github.com/vector-im/element-web/issues/16318 and/or https://github.com/vector-im/element-web/issues/16023)

**Comment by croemheld** [comment]
Until recently I was using the element-desktop application on Ubuntu, along with the Element iOS app. All sessions have been verified. Yesterday I switched to the web version via Riot in Franz,  where my iOS app messages are all marked with the aforementioned shield. This is only happening for all messages written from my iOS app, if I write a message from the desktop app, the message in the web client is not marked with a shield.

**Comment by kittykat** [comment]
> To reproduce, I deleted my secure backup & logged out of all but one session, then logged into a new session & verified via my Security Key. My two sessions trust each other; a DM with another user shows full trust between both users (in both directions), but any messages sent by the other user in the DM appear with gray shields in the new session only. (Messages sent from my own account, from either session, appear with no shield.)

**Comment by kittykat** [comment]
I'm going to close this issue for now:

Duplicate of #16318

**Comment by uhoreg** [comment]
This is actually different from #16318.  16318 is about new messages (messages sent after the device logged in), while this issue is about old messages.

**Comment by foresto** [comment]
Suggestions:

- How about adding an explanation to the warning on those shields? Something like, "This is normal for old messages if you've signed in with a new device." That would have saved my friends a day of frustrating and fruitless troubleshooting.
- If the ultimate goal is to store and recover enough info to avoid this problem, but that can't be implemented soon, perhaps a user's other sessions that *do* still have the missing info could gossip it to the new session? It seems silly that I have one session that can verify old messages and another that can't, given that they already trust and talk to each other.

In any case, thanks very much for explaining what's going on, @uhoreg and @ara4n. For the record, I think the info they represent is important; just presented in a way that needlessly causes alarm.

**Comment by UmbraChimera** [comment]
Is there still not a way to get the "the authenticity of this message can't be confirmed on this device" to go away? I don't know why it's showing up this time but not on any other of my installs I verify.

**Comment by Locne55** [comment]
also noticed this happened when I logged in to a new device

**Comment by BrenBarn** [comment]
> So, for everyone confused by why it's saying this: Normally, in encrypted rooms, messages are authenticated using the decryption key, based on the fact that the decryption key was received directly from the sender. When you sign into a new device (or under some other situations), you may get the decryption key from key backup (keys stored encrypted on the server), or from key sharing (keys sent from your other devices). When that happens, we lose the link to the original sender, and we don't currently store the fact that it was trusted in the first place, which is why it now says that the authenticity can't be verified.

It doesn't have to be verified.  I verified it when I verified the other session.  What it means for me to verify the session is to say "I am stating by fiat that everything this session tells me is true."
 
> So the message is completely expected in certain situations (particularly for old messages when you log in a new device)

It's not expected.  What is expected is that when I verify a session there is no daylight between what that other session thinks is verified and what my new session think is verified.  Their notions of what is verified and trusted should be completely fused.

**Comment by Avamander** [comment]
Even if Element doesn't dare fuse the two together automatically, there should be an option to do it manually.

**Comment by foresto** [comment]
> > So the message is completely expected in certain situations

> It's not expected. 

I think @uhoreg means it's "expected" in the sense that the code is behaving as originally intended, not that the result is good.

This bit of context is important:

> we don't currently store the fact that it was trusted in the first place, which is why it now says that the authenticity can't be verified.

I think (I hope) we can all agree that, although the current behavior makes sense once you understand what's going on behind the scenes, the presentation is misleading, annoying, distracting, and generally makes Element look broken. Fixing it is apparently gated on a design change to store the missing information.

**Comment by BrenBarn** [comment]
> Fixing it is apparently gated on a design change to store the missing information.

Was breaking it in the first place gated on a design change?  It is frustrating to see constantly this "oh we have to deliberate about what to do".  Broken e2ee features have been released, causing problems for users.  Why was the same level of deliberation not given *before* pushing these things out?

**Comment by BloodyIron** [comment]
Yeah I thought this whole "nice UX for e2ee" was solved like two+ years ago? I hear from my family members with regularity they have a really bad UX because they are being "repeatedly" prompted to verify devices (and I think this is just because they are dismissing them, not because they are under threat). And now I see this aspect of e2ee is UX-incomplete?

I agree with @BrenBarn that this feels like it was broken earlier in a way that was avoidable through the rigorous planning that was supposed to have been done (and while I trust was done, this is an area that it fell short).

I want e2ee in Element/Matrix to be "easy" and good for users, and while it is better than before, we're not quite there yet. And this is actually impacting me, someone who is more savvy on such things than my family members, and the "not quite trusted" silver shield has me thinking each time "am I actually secure right now with e2ee or is there something I missed?" and then I discover, no, this is a "bug" of sorts... :/

**Comment by ekg** [comment]
I'd like to echo what others are saying. Basically, if we've authenticated the sessions, we've said trust these keys, and then we've done all these steps to verify the setup, it shouldn't be that you showed some special signal that says that authenticity can't be verified. It is not possible. How can I have verified things but yet not verified them enough? Something isn't working.

**Comment by DemiMarie** [comment]
> There is work in progress to improve the trustworthiness of data of online backups, but the risk still applies to keyshared keys.

What is the status of this?

**Comment by networkException** [comment]
Perhaps the brand new [MSC 4048](https://github.com/uhoreg/matrix-doc/blob/signed_backup/proposals/4048-signed-key-backup.md) is part of that work?

**Comment by uhoreg** [comment]
Yes, MSC4048 should fix this issue.  I'm currently working on an [implementation in the Rust SDK](https://github.com/matrix-org/matrix-rust-sdk/pull/2659).  This, combined with the effort to move the crypto portion of the js-sdk to use the Rust implementation means that this will eventually be solved in Element Web/Desktop.  We're not sure how long it be before it's all ready, but both of those tasks (implementing authenticated backups and getting the Rust integration production-ready) are currently high priority tasks and are being actively worked on by the crypto team.

**Comment by networkException** [comment]
The secret service bug you're describing is a regression introduced in version 1.11.46, see https://github.com/element-hq/element-desktop/issues/1429. I've commented the workaround I've been using to get around the issue, maybe thats helpful to you as well

**Comment by NOP4** [comment]
Hello,
Any update on this?
Is there a way to sign out or reconnect to remove this grey shield?

**Comment by KeenMaron** [comment]
any update on this? switched vom element to element x and got this message for all old and new messages

**Comment by xlyralycanx** [comment]
Still no fix? 😞

**Comment by charludo** [comment]
\+ 1, this is ridiculous.

I've signed into a new phone, and now

- in some (but not all!) chats, OLD messages sent by the OTHER participant have the gray shield;
- in some chats, OLD messages sent by ME have the gray shield;
- in some chats, NEW messages by the OTHER participant have the shield;
- and in some chats NEW messages sent **from the new phone by ME** have the gray shield

Especially the last one is utterly baffling. Also of note: *no* messages have the gray shield in my long-running Element Desktop sessions.

**Comment by James-Mat** [comment]
@charludo as was mentioned above, [MSC 4048](https://github.com/uhoreg/matrix-doc/blob/signed_backup/proposals/4048-signed-key-backup.md) should fix this issue - I had a look over there, and, unfortunately, this seems to not be high priority anymore. It was stated there, that an arrival in the next couple of months is unlikely, so unless some other work supersedes this, it will stay like that for the foreseeable future.

I've been following the development closely and have been eagerly awaiting a state in which I feel like I could suggest Element to less-technical users, but this seems unfortunately still far.

**Comment by ScuolaLaViadiLilith** [comment]
Hello,
we also have this problem on our home server! Everyone who closes one session and opens another (applies on browser, Element X and Element desktop), gets - for encrypted messages exchanged in previous sessions - the grey shield!
Our home server runs Synapse v1.129.0 and Element v1.11.100. For those of us who are not experts, it is a little frustrating to see these grey shields, especially since we are all verified against each other on our home server.
Thank you very much.

**Comment by ScuolaLaViadiLilith** [comment]
We continue to have this problem when using Element X 25.10.2 for iOS, Element Desktop 1.12.3, and Element Web 1.12.2 with Synapse home server v1.141.0.


[issue.txt](https://github.com/user-attachments/files/23342700/issue.txt)
[log-0.log](https://github.com/user-attachments/files/23342703/log-0.log)
[log-1.log](https://github.com/user-attachments/files/23342705/log-1.log)
[log-2.log](https://github.com/user-attachments/files/23342701/log-2.log)
[log-3.log](https://github.com/user-attachments/files/23342706/log-3.log)
[log-4.log](https://github.com/user-attachments/files/23342702/log-4.log)
[log-5.log](https://github.com/user-attachments/files/23342704/log-5.log)
[log-6.log](https://github.com/user-attachments/files/23342715/log-6.log)
[log-7.log](https://github.com/user-attachments/files/23342707/log-7.log)
[log-8.log](https://github.com/user-attachments/files/23342712/log-8.log)
[log-9.log](https://github.com/user-attachments/files/23342713/log-9.log)
[log-10.log](https://github.com/user-attachments/files/23342716/log-10.log)
[log-11.log](https://github.com/user-attachments/files/23342720/log-11.log)
[log-12.log](https://github.com/user-attachments/files/23342710/log-12.log)
[log-13.log](https://github.com/user-attachments/files/23342711/log-13.log)
[log-14.log](https://github.com/user-attachments/files/23342724/log-14.log)
[log-15.log](https://github.com/user-attachments/files/23342727/log-15.log)
[log-16.log](https://github.com/user-attachments/files/23342723/log-16.log)
[log-17.log](https://github.com/user-attachments/files/23342725/log-17.log)
[log-18.log](https://github.com/user-attachments/files/23342729/log-18.log)
[log-19.log](https://github.com/user-attachments/files/23342726/log-19.log)
[log-20.log](https://github.com/user-attachments/files/23342708/log-20.log)
[log-21.log](https://github.com/user-attachments/files/23342709/log-21.log)
[log-22.log](https://github.com/user-attachments/files/23342717/log-22.log)
[log-23.log](https://github.com/user-attachments/files/23342719/log-23.log)
[log-24.log](https://github.com/user-attachments/files/23342728/log-24.log)
[log-25.log](https://github.com/user-attachments/files/23342718/log-25.log)
[log-26.log](https://github.com/user-attachments/files/23342722/log-26.log)
[log-27.log](https://github.com/user-attachments/files/23342714/log-27.log)
[log-28.log](https://github.com/user-attachments/files/23342721/log-28.log)

**Comment by gitvalds** [comment]
I just noticed this problem using Element-web on both Firefox and Chrome, on Fedora 42.
All sessions verified.
All software up to date.

On Chrome I just logged in in incognito window and all the messages, including my own from two minutes before logging in, show the exclamation mark and that message, except for the message sent today from Firefox after logging in on this Chrome.

On Firefox, both private and non-private window, it shows the message for all the messages except for the messages newer than five days ago, which might be the date when I cleared all the cookies and had to log in again.

**Comment by ScuolaLaViadiLilith** [comment]
Hi, I'm still having the same problem! Everyone on my account has this problem, and it's a bit annoying! Hopefully, it gets fixed soon.

[issue.txt](https://github.com/user-attachments/files/27321935/issue.txt)
[log-0.log](https://github.com/user-attachments/files/27321932/log-0.log)
[log-1.log](https://github.com/user-attachments/files/27321934/log-1.log)
[log-2.log](https://github.com/user-attachments/files/27321933/log-2.log)
[log-3.log](https://github.com/user-attachments/files/27321936/log-3.log)

**Comment by Elaws** [comment]
Same issue here after installing Element Nightly for Windows 11. On my main app (Windows 11, not nightly), no issue.

I verified Element Nightly with Element "normal" (with the emojis to compare), etc...

All messages have a "gray shield" in Element Nightly.

Thanks.
